crates/wash-cli/src/call.rs

```diff
@@ -231,14 +227,12 @@ async fn rpc_client_from_opts(
     opts: ConnectionOpts,
     cmd_cluster_seed: Option<ClusterSeed>,
 ) -> Result<(RpcClient, u64)> {
-    let ctx = if let Some(context) = opts.context {
-        Some(load_context(context)?)
-    } else if let Ok(ctx_dir) = context_dir(None) {
-        let ctx_dir = ContextDir::new(ctx_dir)?;
-        ensure_host_config_context(&ctx_dir)?;
-        Some(ctx_dir.load_default_context()?)
+    // Attempt to load a context, falling back on the default if not supplied
+    let ctx_dir = ContextDir::new()?;
+    let ctx = if let Some(context_name) = opts.context {
+        Some(ctx_dir.load_context(&context_name)?)
     } else {
-        None
+        ctx_dir.load_default_context().ok()
```

Comment:
NIT: `ctx_dir.load_context` and `load_default_context` have the same `Result<WashContext>` -- do we need to wrap in `Some` if we know it will always be there? seems like a bunch of code after this could be simplified as well if we know we always load either the given context or the default one.

Also, it would probably be nice to add a `.context("failed to load context @ [...]")` or similar for failures here.

This isn't critical though, biased for :racehorse: here.

Reply:
Nope, you're totally right. Didn't think of this during late-night refactoring 😴